### ISSUE TITLE
`IsLoggedIn` should probably be true when connecting

### ISSUE THREAD

**Issue by peppy** [ppy/osu#14393]
Noticed in https://github.com/ppy/osu/pull/14392

`IsLoggedIn` doesn't really expose correct state during the connection process. It should probably still be `true` even when `APIState.Connecting` and potentially also `APIState.Failing`, else messaging to the user is incorrect in some places (ie. the leaderboard at song select says "please login" even though they are logged in). Will open an issue to track this, but also left a TODO as it's kind of important to know this when using that flag.

**Comment by smoogipoo** [comment]
I agree but only if the user has already been in an `Online` state. `Connecting` to me doesn't imply having the correct credentials, which is what being "logged in" does.
